Library/Homebrew/service.rb

```diff
@@ -28,14 +28,48 @@ def initialize(formula, &block)
       @run_type = RUN_TYPE_IMMEDIATE
       @run_at_load = true
       @environment_variables = {}
-      @service_block = block
+      instance_eval(&block) if block
     end
 
     sig { returns(Formula) }
     def f
       @formula
     end
 
+    sig { returns(String) }
+    def default_plist_name
+      "homebrew.mxcl.#{@formula.name}"
+    end
+
+    sig { params(name: T.nilable(String)).returns(String) }
+    def plist_name(name = nil)
+      case T.unsafe(name)
+      when nil
+        @plist_name ||= default_plist_name
+      when String
+        @plist_name = name
+      else
+        raise TypeError, "Service#plist_name expects a String"
+      end
+    end
+
+    sig { returns(String) }
+    def default_service_name
+      "homebrew.#{@formula.name}"
+    end
+
+    sig { params(name: T.nilable(String)).returns(String) }
+    def service_name(name = nil)
+      case T.unsafe(name)
+      when nil
+        @service_name ||= default_service_name
+      when String
+        @service_name = name
+      else
+        raise TypeError, "Service#service_name expects a String"
+      end
+    end
```

Comment:
so is `plist_name` only ever used on macOS and `service_name` only ever used on Linux? If so, I wonder if `macos_plist_name` and `linux_service_name` or something similar would be better and make this more obvious. We can't really change the `Formula` DSL here (although we could deprecate it and rename it that way) but I'd love the new names to be as good as they can be.

Reply:
I think it'd be weird to have different names that refer to the same thing inside the DSL. Can't this just be solved with documentation? I agree that if we could do it all over again maybe we'd pick different names though (but isn't this a common thought when programming).

Reply:
I don't think it'd be a good idea to allow users to specify the file extension but other than that I really like this idea. There are some edge cases cases I'd have to figure out but this sounds good.

Reply:
Fine with me as long as we have checks on not providing the extension.

Reply:
@MikeMcQuaid I believe it is `systemd` on Linux and `launchd` on MacOS. Internally, we don't check for the OS though. We just check for the `systemctl` and `launchctl` commands. There seem to be some Linux distros that don't use `systemd` though.

Reply:
Yeah, we could only really check the OS for launchd, otherwise we'd break on Linux distributions that don't use systemd.

Technically if you have systemd installed on macos you could use service_name there.

Reply:
> Fine with me as long as we have checks on not providing the extension.

@carlocab I'm not sure that's really worth the effort here. How about we wait for now and add it in later if we're seeing problems from users?